addons/auth_totp/models/ir_http.py

```diff
@@ -1,13 +0,0 @@
-# -*- coding: utf-8 -*-
-from odoo import models
-from odoo.http import request
-
-class IrHttp(models.AbstractModel):
-    _inherit = 'ir.http'
-
-    def session_info(self):
-        info = super().session_info()
-        # because frontend session_info uses this key and is embedded in
-        # the view source
-        info["user_id"] = request.session.uid,
```

Comment:
Even if this fix was correct, you wouldn't be able to remove this in a stable version: it's guaranteed to break third-party code that relies on this key.

That's an important rule of the stable policy: never remove something that third-party code may have started to rely on. Especially something very clearly exposed like a key in `session_info` ;)

Reply:
@yankinmax I've added more info/analysis on a duplicate PR here: https://github.com/odoo/odoo/pull/96576#issuecomment-1195709272 

For 15.0 I believe the only safe option is to fix the few pieces of code that use `getSession().user_id[0]` in the backend, and replace that with `getSession().uid`. Other changes to unify/cleanup the 2 keys (uid/user_id) should be done in master.

Reply:
NP, we can close this PR after changes introduced in duplicate PR will be merged. Thanks